### ISSUE TITLE
Top statistics per language and project

### ISSUE THREAD

**Issue by comradekingu** [WeblateOrg/weblate#2752]
Right now statistics are only available from the dashboard, and they cover the whole instance.
For consistency, building community within a language, and the usefulness of reaching out to translators when something needs to be done urgently, it would be nice to see statistics in the language tab, and also per project.

The call for also having per project based statistics went something like this:

Lets say someone wants to reach out to the top translators for any given project. It could be that there are a lot of other translators, that just translate other projects, and have no interest in translating the one project in question.

**Comment by nijel** [comment]
The problem is that we currently do not collect such information.

**Comment by comradekingu** [comment]
Would it be very costly in terms of computation?
The attribution plugin is a bit similar, no?

**Comment by comradekingu** [comment]
@nijel In my head it is generated data, a run each day is fine.
https://hosted.weblate.org/stats/ has
(total amount of) translatable strings and
(total amount of) translatable words

https://hosted.weblate.org/languages has
(current amount of) translated strings and (though in %)
(current amount of) translated words (though in %)

It would be nice to have current in % completion of total, in two extra columns in https://hosted.weblate.org/languages, and/or make both current columns sortable by total completion.

![bilde](https://user-images.githubusercontent.com/13802408/58926999-4c122580-874d-11e9-8919-e50a8c37372c.png)

**Comment by nijel** [comment]
@comradekingu I guess your last comment belongs to https://github.com/WeblateOrg/weblate/issues/2732

**Comment by comradekingu** [comment]
I forgot I had posted that, but remembered this one that I forwarded for someone else. :)

**Comment by comradekingu** [comment]
@nijel 
I unconfused myself. The idea with this is to have the same top statistics which are now instance wide, per language, and project. Especially per language I would want to know who the top translators are, to build community, and also trust by checking each individual one. That seems to be the simplest and most expected approach.
The list could be longer than it is now, but that is a bonus.

The other way to do something similar is to have a main category to click, being "Translators" with similar bars to what projects and languages has now. That however complicates bars between views, and also touches on this issue:

Nothwithstanding checks, is possible to go

- Project/language/component or

- Project/compoent and then select language

or

- Language/project/component

or get to the

- Translator view

but going between every combination of the former, means views have to be generated on the fly(?), with some logic to click things in and out of view.

**Comment by GravuTrad** [comment]
I'm agree with the request of this enhancement. Very useful to have datas for each work done by language.

**Comment by rajasekharponakala** [comment]
I'm planning to work on this requirement.

I observed that from https://hosted.weblate.org/stats/, these below blocks of stats are already available in https://hosted.weblate.org/languages/*/#information & https://hosted.weblate.org/projects/*/#information

- String statistics
- Activity in last 12 months
- Quick numbers
- Trends of last 30 days

Only 'Most active translators' is yet to be added for both language and project info pages. Please clarify this, so I can start working on this.

**Comment by nijel** [comment]
Yes, the most active translators are not available at other scopes. The biggest issue in this is to collect such data reasonably fast - site wide we count the contributors per user, and it's cheap to get users ordered by number of contributions, but there are currently no such aggregates at other levels. Probably, it could be incorporated into the metrics framework, but it needs to be implemented.

**Comment by comradekingu** [comment]
In a similar vein, the number of users with the language, or even only that language selcted is known,
which probably accounts for most all contributions…?
Doesn't have to be super accurate to tell a story. People don't generally change it all that often either I would think.

If such lists are generated, the latest members of any language ala https://hosted.weblate.org/user/
could be listed in the lower right corner of 
![image](https://github.com/WeblateOrg/weblate/assets/13802408/e74d4c18-3717-4d97-a54c-33bbd4335448)

@rajasekharponakala How are you getting on? :)
